inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,500 @@
+    std::vector<std::shared_ptr<fims::Fleet<Type> > > fleets; /*!< shared pointer to fleet module */
+
+    // this -> means you're referring to a class member (member of self)
+
+    // std::vector<std::shared_ptr<fims::Fleet<Type>> > surveys;
+    Population()
+    {
+      this->id = Population::id_g++;
+    }
+
+    /**
+     * @brief Initialize values. Called once at the start of model run.
+     *
+     * @param index_ya dimension folded index for year and age
+     * @param age age index
+     */
+    void Initialize(int nyears, int nseasons, int nages)
+    {
+      // size all the vectors to length of nages
+      nfleets = fleets.size();
+      ages.resize(nages);
+      catch_numbers_at_age.resize(nyears * nages * nfleets);
+      mortality_F.resize(nyears * nages);
+      mortality_Z.resize(nyears * nages);
+      proportion_mature_at_age.resize(nyears * nages);
+      weight_at_age.resize(nages);
+      catch_weight_at_age.resize(nyears * nages * nfleets);
+      unfished_numbers_at_age.resize((nyears + 1) * nages);
+      numbers_at_age.resize((nyears + 1) * nages);
+      expected_catch.resize(nyears * nfleets);
+      expected_index.resize(nyears * nfleets);
+      biomass.resize((nyears + 1));
+      unfished_spawning_biomass.resize((nyears + 1));
+      spawning_biomass.resize((nyears + 1));
+      log_naa.resize(nages);
+      log_Fmort.resize(nfleets * nyears);
+      log_M.resize(nyears * nages);
+      log_q.resize(nfleets);
+      naa.resize(nages);
+      Fmort.resize(nfleets * nyears);
+      M.resize(nyears * nages);
+      q.resize(nfleets);
+    }
+
+    /**
+     * @brief Prepare to run the population loop. Called at each model itartion, and used
+     *  to zero out derived quantities, values that were summed, etc.
+     *
+     */
+    void Prepare()
+    {
+      std::fill(unfished_spawning_biomass.begin(), unfished_spawning_biomass.end(), 0);
+      std::fill(spawning_biomass.begin(), spawning_biomass.end(), 0);
+      std::fill(mortality_F.begin(), mortality_F.end(), 0);
+      std::fill(expected_catch.begin(), expected_catch.end(), 0);
+
+      // Transformation Section
+      for (size_t age = 0; age < this->nages; age++)
+      {
+        // for (size_t age = 0; age <= this->nages; age++) {
+        // Use fims::exp?
+        this->naa[age] = fims::exp(this->log_naa[age]);
+        for (size_t year = 0; year < this->nyears; year++)
+        {
+          int index_ya = year * this->nages + age;
+          this->M[index_ya] = fims::exp(this->log_M[index_ya]);
+        }
+      }
+
+      for (size_t fleet_ = 0; fleet_ < this->nfleets; fleet_++)
+      {
+        // for(size_t fleet_ = 0; fleet_ <= this->nfleets; fleet_++) {
+        // this -> Fmort[fleet_] = fims::exp(this -> log_Fmort[fleet_]);
+        for (size_t year = 0; year < this->nyears; year++)
+        {
+          int index_yf = year * this->nfleets + fleet_;
+          this->Fmort[index_yf] = fims::exp(this->log_Fmort[index_yf]);
```

Comment:
@Andrea-Havron-NOAA asked on the other PR about whether treating surveys as fleets means we will have an F for surveys. @cmlegault reminded me the issue with treating surveys as fleets is that then you need to account for surveys in allocation, and I'm not clear if this will affect our milestone 1 integration test or only subsequent milestones

Reply:
moved to issue